verifier/verifier.go

```diff
@@ -518,6 +549,76 @@ func verifyAuthenticTimestamp(outcome *notation.VerificationOutcome) *notation.V
 	}
 }
 
+func verifyRevocation(outcome *notation.VerificationOutcome, r revocation.Revocation, logger log.Logger) *notation.ValidationResult {
+	if r == nil {
+		return &notation.ValidationResult{
+			Type:   trustpolicy.TypeRevocation,
+			Action: outcome.VerificationLevel.Enforcement[trustpolicy.TypeRevocation],
+			Error:  fmt.Errorf("unable to check revocation status, revocation client cannot be nil"),
+		}
+	}
+
+	authenticSigningTime, err := outcome.EnvelopeContent.SignerInfo.GetAuthenticSigningTime()
+	if err != nil {
+		logger.Debugf("not using authentic signing time due to error retrieving AuthenticSigningTime, err: %v", err)
+	}
+	revocationResult, err := r.Validate(outcome.EnvelopeContent.SignerInfo.CertificateChain, authenticSigningTime)
+	if err != nil {
+		logger.Debug("error while checking revocation status, err: %s", err.Error())
+		return &notation.ValidationResult{
+			Type:   trustpolicy.TypeRevocation,
+			Action: outcome.VerificationLevel.Enforcement[trustpolicy.TypeRevocation],
+			Error:  fmt.Errorf("unable to check revocation status, err: %s", err.Error()),
+		}
+	}
+
+	result := &notation.ValidationResult{
+		Type:   trustpolicy.TypeRevocation,
+		Action: outcome.VerificationLevel.Enforcement[trustpolicy.TypeRevocation],
+	}
+
+	finalResult := revocationresult.ResultUnknown
+	numOKResults := 0
+	var problematicCertSubject string
+	revokedFound := false
+	var revokedCertSubject string
+	for i := len(revocationResult) - 1; i >= 0; i-- {
+		if revocationResult[i].ServerResults[0].Error != nil {
+			logger.Debugf("error for certificate #%d in chain with subject %v for server %q: %v", (i + 1), outcome.EnvelopeContent.SignerInfo.CertificateChain[i].Subject.String(), revocationResult[i].ServerResults[0].Server, revocationResult[i].ServerResults[0].Error)
+		}
+
+		if revocationResult[i].Result == revocationresult.ResultOK || revocationResult[i].Result == revocationresult.ResultNonRevokable {
+			numOKResults++
+		} else {
+			finalResult = revocationResult[i].Result
+			problematicCertSubject = outcome.EnvelopeContent.SignerInfo.CertificateChain[i].Subject.String()
+			if revocationResult[i].Result == revocationresult.ResultRevoked {
+				revokedFound = true
+				revokedCertSubject = problematicCertSubject
+			}
+		}
+	}
+	if revokedFound {
+		problematicCertSubject = revokedCertSubject
+		finalResult = revocationresult.ResultRevoked
+	}
+	if numOKResults == len(revocationResult) {
+		finalResult = revocationresult.ResultOK
+	}
+
+	switch finalResult {
+	case revocationresult.ResultOK:
+		logger.Debug("no important errors encountered while checking revocation, status is OK")
```

Comment:
I'm not sure what does "no `important` errors" mean here? Should it just be `no error`?
```suggestion
		logger.Debug("no error encountered while checking revocation, status is OK")
```

Reply:
Updated to "no verification impacting errors" rather than important in order to provide more clarity.